docs/api/v1.md

```diff
@@ -106,12 +292,89 @@ to stream trace with chunked encoding, maybe omit status).
 }
 ```
 
+
 ### Execution
 ```
 {
-    id {int}: Unique Identifier
-    otherInfo {object}: TODO Whatever other info, ask Russell 
+   id {integer}: Unique Identifier
+   type {string}: Type of execution. Either `function` or `process`
+   status {string}: Either `pending`, `executing`, `stopped`, or `done`
+   executionTime {integer}: Nanoseconds
+   data {object}: Data specific to this type
+      if `type=function`: {
+         function {[Function](#function)}: Function that produced this execution
+      }
+      if `type=process`: {
+         nextExecution {Execution id}: Id of the execution that follows this one. Null until `status=done`. Initially set to `null`, set when a call to `/debug/execute` hits a breakpoint, in which case it points to the breakpoint function�s execution.
+      }
 }
 ```
 
+
+### Trace
+```
+{
+   index {integer}: Index of the trace, beginning at 0 and totally ordered for each execution
+   type {integer}: Type of trace: `0`=instruction (state changes), `1`=output (to stdout), `2`=termination (data about how the execution ended)
+   line {integer}: Line number that produced this trace
+   data {object}: Data specific to the trace type
+      if `type=0`: {
+         state {object[]}: Array of variable changes {variable {Variable}, value {*}}
+      }
+      If `type=1`: {
+         output {string}: Stdout output
+      }
+      If `type=2`: {
+         cause {string}: Reason for termination. Either `stopped`, `crashed`, `ended`, or `breakpoint`
+         stack {string} (defined if cause=crashed): Stack trace output
+         code {integer} (defined if cause=ended): Program exit code
+         nextExecution {Execution id} (defined if cause=breakpoint): Id of the following function execution
+      }
+}
+```
+
+
+### Function
+```
+{
+   address {integer}: Memory address of the function
+   name {string}: Name of the function
+   sourcePath {string}: Full path to the source code
+   lineNumber {integer}: Line number of the function definition
```

Comment:
Maybe include a line number range or file offset range so we can display just the function?